Lib/mastermind.rb

```diff
@@ -0,0 +1,59 @@
+class Mastermind
+  def show_instructions
+    puts 'Welcome to Mastermind! Your challenge is
+to guess the secret four digit code in under 10 tries.
+
+The code can include the following colors:
+Red(r), Green(g), Orange(o), Blue(b), Yellow(y) and Purple(p)
+Enter your guess by entering four color values from above, such as rgyb or pggy.
+
+If you have a correct color in a correct position, you get a white pin.
+If you have a correct color in a different position, you get a black pin
+press \'q\' to quit, or enter your first guess.'
+  end
+end
+
+class Code
+  attr_reader :code
+
+  def generate_code
+    code = []
+    @colors = ["r", "g", "o", "b", "y", "p"]
+    4.times do |inject|
+      code << @colors.sample
+    end
+    code
+  end
+end
+
+class Response
+  attr_reader :player_input
+
+  def initialize
+    @valid_guess = false
+    @player_input = ""
+  end
+
+  def get_input
+    puts "Please enter a guess from the available colors: r, g, o, b, y, p"
+    @player_input = gets.chomp
+    return @player_input.downcase.gsub(/[\W]/, "")
+  end
+
+  def validate_length(guess)
```

Comment:
Right now this method does 2 things. It checks whether the length is valid, and prints to standard out if it's not. What if we rename this method to `valid_length?`, and have it just return `@player_input.length != 4`. That way, in a separate method we can do something like:
```rb
if valid_length?(guess)
  puts "Invalid response, please enter exactly 4 colors"
end
```
Now the `valid_length?` method becomes much easier to test because you're just testing whether it returns `true` or `false`, which is that the actual logic we're looking to test, instead of what it prints to stdout.